text/0000-result-expect.md

```diff
@@ -0,0 +1,40 @@
+- Feature Name: `result_expect`
+- Start Date: 2015-05-13
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Add an `expect` method to the Result type, bounded to `E: Debug`
+
+# Motivation
+
+While `Result::unwrap` exists, it does not allow annotating the panic message with the operation
+attempted (e.g. what file was being opened). This is at odds to 'Option' which includes both
+`unwrap` and `expect` (with the latter taking an arbitrary failure message).
+
+# Detailed design
+
+Add a new method to the same `impl` block as `Result::unwrap` that takes a `&str` message and
```

Comment:
Ooh, interesting idea. This would allow passing `format_args!` for formatted messages